dfvfs/lib/gzipfile.py

```diff
@@ -152,41 +152,71 @@ def __init__(
     self.operating_system = None
     self.original_filename = None
 
+    # Read the member data to determine the uncompressed data size and
+    # the offset of the member footer.
+    file_size = file_object.get_size()
+
+    file_object.seek(member_start_offset, os.SEEK_SET)
+    self._ReadMemberHeader(file_object)
+
+    uncompressed_data_size = 0
+
+    compressed_data_offset = file_object.get_offset()
+    decompressor_state = _GzipDecompressorState(compressed_data_offset)
+
+    file_offset = compressed_data_offset
+    while file_offset < file_size:
+      decompressed_data = decompressor_state.Read(file_object)
```

Comment:
This means we read the whole member without priming the cache - could we at least cache all this if we knew the uncompressed data size is smaller than the maximum buffer size? What's the rationale for not priming the cache as we go?

Reply:
I've made some changes to cache gzip files that fit in the cache

Reply:
the reason for not priming the cache in other situations is that it will point to end of file and will very likely be invalidated by the next read.